src/dragger.js

```diff
@@ -0,0 +1,126 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Multiple selection dragger.
+ */
+
+import * as Blockly from 'blockly/core';
+import {blockSelection, inMultipleSelectionMode, hasSelectedParent, origHighlight} from './global';
+
+/**
+ * A block dragger that adds the functionality for multiple block to
+ * be moved while someone is dragging it.
+ */
+export class MultiSelectBlockDragger extends Blockly.BlockDragger {
+  /** @override */
+  constructor(block, workspace) {
+    super(block, workspace);
+    this.block_ = block;
+    this.workspace_ = workspace;
+    this.group_ = '';
+    this.blockDraggers_ = new Set();
+  }
+
+  /**
+   * Prepares the block dragger for a new drag.
+   * @param {!Blockly.utils.Coordinate} currentDragDeltaXY How far the pointer
+   *     has moved from the position at the start of the drag, in pixel units.
+   * @param {boolean} healStack Whether or not to heal the stack after
+   *     disconnecting.
+   */
+  startDrag(currentDragDeltaXY, healStack) {
+    if (!inMultipleSelectionMode) {
+      const blockDraggerList = [];
+      if (!blockSelection.has(this.block_.id)) {
+        blockSelection.forEach((id) => {
+          const element = this.workspace_.getBlockById(id);
+          if (!element.disposed) {
```

Comment:
What case is this checking for? I think that if the block is disposed, it won't be returned by `getBlockById`

Reply:
Oh, I used to directly store the BlockSVG obj into the blockSelection set, these are old code that didn't get cleaned up, so will change into directly checking the element (`if (element)`).